src/messages/accumulating.rs

```diff
@@ -31,18 +34,22 @@ pub struct AccumulatingMessage {
 impl AccumulatingMessage {
     /// Create new `AccumulatingMessage`
     pub fn new(
-        content: RoutingMessage,
+        src: Prefix<XorName>,
+        dst: DstLocation,
+        variant: Variant,
         section_share: &SectionKeyShare,
         public_key_set: bls::PublicKeySet,
         proof: SectionProofSlice,
     ) -> Result<Self> {
-        let bytes = serialize(&content)?;
+        let bytes = serialize(&(&src, &dst, &variant))?;
```

Comment:
This split of one type into multiple does not seem quite right, especially as we serialize it like that with a specific order in multiple places. Would it make sense to still have a type, maybe one that serialize like it happens here?